### PR TITLE
[BUGFIX][MER-2670] Soft scheduled cant be removed

### PR DESCRIPTION
**Pull request by simonchoxx**
[MER-2670](https://eliterate.atlassian.net/browse/MER-2670)

This PR adds the ability to clear the timelines of a schedule. 

If you have set timelines you will be able to clean them completely, which until now was not possible.

https://github.com/Simon-Initiative/oli-torus/assets/16328384/23ba3dad-ad12-406b-b624-5604d8d682fa

[MER-2670]: https://eliterate.atlassian.net/browse/MER-2670?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/oli_web/controllers/api/scheduling_controller_test.exs
Comment (gastonabella) [inline test/oli_web/controllers/api/scheduling_controller_test.exs:118]: Nit but you can use `refute` here:

```elixir
 refute sr1["start_date"]
 ...
```

**Review by gastonabella — COMMENTED** [review]
This looks good! 🔥
Just left some small comments.
Besides that, I'd like to bring up a couple of styling improvements (which are just an opinion and I'm ok with leaving them as they are too 🙂):
- Both `Reset Timelines` and `Clear Timelines` buttons are IMO too close to the table below. Some bottom margin would look better. Also, both are buttons but look like links (specially because of the underline).
- When clearing the timelines, the `Ok` and `Cancel` buttons are too close to the text right above.

File: lib/oli_web/controllers/api/scheduling_controller.ex
Comment (simonchoxx) [reply lib/oli_web/controllers/api/scheduling_controller.ex:179]: You are right!

**Comment by darrensiegel** [comment]
Hold, do not merge for V26.

File: lib/oli_web/controllers/api/scheduling_controller.ex
Comment (simonchoxx) [reply lib/oli_web/controllers/api/scheduling_controller.ex:183]: I agree, but.. what scenario do you think should handle some kind of error that occurs when doing `Repo.Update_all`?

Comment (gastonabella) [reply lib/oli_web/controllers/api/scheduling_controller.ex:183]: According to the docs, [update_all](https://hexdocs.pm/ecto/Ecto.Repo.html#c:update_all/3) will never return an error, even if it fails. I guess one way of knowing whether an insert failed or not is pattern matching on `{0, _}`, since 0 records inserted could mean that the operation failed, but we are not 100% about that.

File: test/oli_web/controllers/api/scheduling_controller_test.exs
Comment (gastonabella) [reply test/oli_web/controllers/api/scheduling_controller_test.exs:118]: Good question! 
`refute` is defined to ensure an expression is `false` or `nil`, which is the case, but I agree that asserting against a specific value can be more explicit, so I leave it up to you! I'm ok with either way :)

**Review by bitbldr — APPROVED** [review]
Approved for v27 forward development

**Pull request merged**
